crates/anvil-polkadot/src/substrate_node/service.rs

```diff
@@ -134,30 +194,162 @@ pub fn new(anvil_config: &AnvilNodeConfig, config: Configuration) -> Result<Serv
     })
 }
 
-fn spawn_rpc_server(
-    task_manager: &mut TaskManager,
+// Re-implement RPC module generation without the check on the genesis block number.
+// The code is identical to the one in
+// https://github.com/paritytech/polkadot-sdk/blob/9e0636567bebf312b065ca3acb285a8b32499df7/substrate/client/service/src/builder.rs#L754
+// apart from the creation of the RPC builder inside the function and the genesis number check.
+#[allow(clippy::too_many_arguments)]
+fn custom_gen_rpc_module(
```

Comment:
let's just call it gen_rpc_module

Reply:
nit: we could move these two functions in a separate `rpc` file under substrate_node